registry/registry.go

```diff
@@ -133,9 +114,11 @@ func ReadCertsDirectory(tlsConfig *tls.Config, directory string) error {
 // DockerHeaders returns request modifiers that ensure requests have
 // the User-Agent header set to dockerUserAgent and that metaHeaders
 // are added.
-func DockerHeaders(metaHeaders http.Header) []transport.RequestModifier {
+func DockerHeaders(userAgent string, metaHeaders http.Header) []transport.RequestModifier {
 	modifiers := []transport.RequestModifier{
-		transport.NewHeaderRequestModifier(http.Header{"User-Agent": []string{dockerUserAgent}}),
+		transport.NewHeaderRequestModifier(http.Header{
+			"User-Agent": []string{userAgent},
```

Comment:
I think I should have elaborated on my previous comment. No I do not think `userAgent == ""` should be considered an error. But when an empty user agent is given it should not set the header value and should let Go use the default user agent. When an empty User-Agent is given, Go will not send any user agent, see https://golang.org/src/net/http/request.go#L419.